### PR TITLE
utils: Use an extra drainer for slow logs

### PR DESCRIPTION
**Pull request by Fullstop000**
Signed-off-by: Fulstop000 <fullstop1005@gmail.com>

UCP #6735

###  What have you changed?

Output the slow log to a dedicated file

###  What is the type of the changes?

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?

- Unit test
- Integration test

### CODE REVIEW

**Comment by Fullstop000** [comment]
@overvenus PTAL

File: src/config.rs [hunk 1]
```diff
@@ -1771,6 +1771,9 @@ pub struct TiKvConfig {
     #[config(skip)]
     pub log_file: String,
 
+    #[config(skip)]
+    pub slow_log_file: String,
```
Comment (overvenus) [inline src/config.rs]: Could you update `etc/config-template.toml` and `test_serde_custom_tikv_config?

Comment (overvenus) [inline src/config.rs]: Could you support setting slow-threshold just like TiDB?

https://github.com/pingcap/tidb/blob/1771cf8870f999fa229552e9a3034d166708fbad/config/config.go#L217

Comment (Fullstop000) [reply src/config.rs]: done

**Comment by Fullstop000** [comment]
ping @overvenus

**Comment by Fullstop000** [comment]
@sre-bot /run-all-tests

**Comment by Fullstop000** [comment]
@sre-bot /run-all-tests

**Comment by Fullstop000** [comment]
@sre-bot /bench

**Comment by TennyZhuang** [comment]
@sre-bot /bench

**Comment by Fullstop000** [comment]
@sre-bot /bench

File: components/tikv_util/src/logger/mod.rs [hunk 2]
```diff
@@ -304,6 +384,23 @@ fn write_log_fields(
...
 
+struct SlowCostSerializer {
+    // None means input record without key `takes`
+    cost: Option<u64>,
+}
+
+impl slog::ser::Serializer for SlowCostSerializer {
+    fn emit_arguments(&mut self, key: Key, val: &fmt::Arguments<'_>) -> slog::Result {
+        if key == "takes" {
+            match format!("{}", val).parse() {
```
Comment (overvenus) [inline components/tikv_util/src/logger/mod.rs]: It seems like parse the value for every slow log?

**Comment by Fullstop000** [comment]
@sre-bot /run-all-tests

**Comment by Fullstop000** [comment]
@sre-bot /run-all-tests

**Pull request merged**